packages/synthetix-main/contracts/interfaces/IRewardsManagerModule.sol

```diff
@@ -20,11 +19,17 @@ interface IRewardsManagerModule {
         uint amountClaimed
     );
 
+    /// @notice called by pool owner or an existing distributor to register rewards for vault participants
+    function registerRewardsDistributor(
+        uint128 poolId,
+        address token,
+        address distributor
+    ) external;
+
     /// @notice called by pool owner or an existing distributor to set up rewards for vault participants
-    function setRewardsDistribution(
+    function distributeRewards(
         uint128 poolId,
         address token,
-        uint index,
         address distributor,
```

Comment:
I think this isn't needed. Could use `msg.sender` and then check that a rewards distributor has been registered with that address.